lib/faraday/connection.rb

```diff
@@ -79,15 +79,22 @@ def initialize(url = nil, options = nil)
       @params.update(options.params)   if options.params
       @headers.update(options.headers) if options.headers
 
-      @proxy = @no_proxy = nil
+      @proxy = nil
       proxy(options.fetch(:proxy) {
-        uri = ENV['http_proxy']
-        if uri && !uri.empty?
-          uri = 'http://' + uri unless uri.downcase.start_with?('http')
-          uri
+        uri = nil
+        if URI.parse("").respond_to?(:find_proxy)
```

Comment:
>  Since we dropped the support for older ruby version, I think this method should always be available?

`open-uri` lib has `find_proxy` in >= Ruby 1.8.7. This method move to `uri/generic` lib in Ruby 2.0.0.
I think any library shouldn't require `open-uri` because this library overwrite Kernel#.open.
So this pull request doesn't support no_proxy in Ruby 1.9.3. 😢